### PR TITLE
PostShare: split component

### PR DESCRIPTION
**Pull request by retrofox**
This PR splits up `<PostShare />` component creating a new `<PublicizeActionsList />`.

### Testing

Test the whole post share section. Everything should be fine.

### CODE REVIEW

File: client/my-sites/post-share/index.jsx [hunk 1]
```diff
@@ -42,55 +37,52 @@ import {
...
 	};
 
-	static FOOTER_SECTION_SCHEDULED = 'footer-section-scheduled';
-	static FOOTER_SECTION_PUBLISHED = 'footer-section-published';
+	static defaultProps = {
+		connections: [],
+	};
 
 	state = {
+		footerSection: FOOTER_SECTION_SCHEDULED,
```
Comment (gwwar) [inline client/my-sites/post-share/index.jsx]: Maybe `selectedShareTab`, with a value of 'published' or 'scheduled'

File: client/my-sites/post-share/index.jsx [hunk 2]
```diff
@@ -42,55 +37,52 @@ import {
...
 		skipped: PostMetadata.publicizeSkipped( this.props.post ) || [],
-		footerSection: PostShare.FOOTER_SECTION_SCHEDULED
-
 	};
 
 	setFooterSection = footerSection => () => this.setState( { footerSection } );
 
 	hasConnections() {
-		return !! ( this.props.connections && this.props.connections.length );
+		return !! this.props.connections.length;
```
Comment (gwwar) [inline client/my-sites/post-share/index.jsx]: connections can be falsey I think, if you'd like to shorten this, let's use lodash get

Comment (retrofox) [inline client/my-sites/post-share/index.jsx]: should it be handled by the default value? Let's add the `get` selector anyway.

File: client/my-sites/post-share/index.jsx (hunk 1 above)
Comment (retrofox) [inline client/my-sites/post-share/index.jsx]: 02e65416b35b353f3b7a79814dfaf83224923eb6

File: client/my-sites/post-share/index.jsx (hunk 2 above)
Comment (retrofox) [inline client/my-sites/post-share/index.jsx]: f6c6b2c2924eb11df45d5d03d899dbdec16f83e3

File: client/my-sites/post-share/index.jsx [hunk 3]
```diff
@@ -203,121 +195,75 @@ class PostShare extends Component {
...
+				] }
+				plan={ PLAN_BUSINESS }
+				title={ translate( 'Upgrade to a Business Plan!' ) }
+			/>
+		);
 	}
 
 	renderFooter() {
-		const { footerSection } = this.state;
+		if ( ! isEnabled( 'publicize-scheduling' ) ) {
```
Comment (gwwar) [inline client/my-sites/post-share/index.jsx]: Shouldn't we also check for business here?

Comment (retrofox) [inline client/my-sites/post-share/index.jsx]: 2955df0496e500e7480cbfed8d339920b4427121 and bd284ccc38375f964239ff5d26ecc4f1fc7a9ba9

File: client/my-sites/post-share/constants.js [hunk 4]
```diff
@@ -0,0 +1,3 @@
+
+export const SECTION_SCHEDULED = 'scheduled';
```
Comment (gwwar) [inline client/my-sites/post-share/constants.js]: How about: 

```
export const SCHEDULED = 'scheduled`;
export const PUBLISHED = 'published';
```

Comment (retrofox) [inline client/my-sites/post-share/constants.js]: fe546fc7de4daa56859af690f139ff02a31bc0cf

**Comment by retrofox** [comment]
rebasing and merging. Thanks!

**Pull request merged**